src/qibo/backends/__init__.py

```diff
@@ -78,39 +78,73 @@ class GlobalBackend(NumpyBackend):
         {"backend": "pytorch"},
     ]
 
-    def __new__(cls):
-        if cls._instance is not None:
-            return cls._instance
+    @classmethod
+    def backend(cls):  # pragma: no cover
+        if cls._backend is not None:
+            return cls._backend
 
         backend = os.environ.get("QIBO_BACKEND")
-        if backend:  # pragma: no cover
+        if backend:
             # Create backend specified by user
             platform = os.environ.get("QIBO_PLATFORM")
-            cls._instance = construct_backend(backend, platform=platform)
+            cls._backend = construct_backend(backend, platform=platform)
         else:
             # Create backend according to default order
             for kwargs in cls._default_order:
                 try:
-                    cls._instance = construct_backend(**kwargs)
+                    cls._backend = construct_backend(**kwargs)
                     break
-                except (ImportError, MissingBackend):
+                except (ModuleNotFoundError, ImportError):
                     pass
 
-        if cls._instance is None:  # pragma: no cover
+        if cls._backend is None:
             raise_error(RuntimeError, "No backends available.")
 
-        log.info(f"Using {cls._instance} backend on {cls._instance.device}")
-        return cls._instance
+        log.info(f"Using {cls._backend} backend on {cls._backend.device}")
+        return cls._backend
 
     @classmethod
     def set_backend(cls, backend, **kwargs):  # pragma: no cover
         if (
-            cls._instance is None
-            or cls._instance.name != backend
-            or cls._instance.platform != kwargs.get("platform")
+            cls._backend is None
+            or cls._backend.name != backend
+            or cls._backend.platform != kwargs.get("platform")
         ):
-            cls._instance = construct_backend(backend, **kwargs)
-        log.info(f"Using {cls._instance} backend on {cls._instance.device}")
+            cls._backend = construct_backend(backend, **kwargs)
+            log.info(f"Using {cls._backend} backend on {cls._backend.device}")
+        else:
+            log.info(f"Backend {backend} is already loaded.")
```

Comment:
I wonder whether we should stop logging, since this is quite unexpected when Qibo is used as a library within other applications (or even other libraries).

But this is going beyond this PR.

Reply:
I changed it so that it only prints when the backend is set.